langchain4j-weaviate/src/test/java/dev/langchain4j/store/embedding/weaviate/LocalGRPCWeaviateEmbeddingStoreIT.java

```diff
@@ -0,0 +1,74 @@
+package dev.langchain4j.store.embedding.weaviate;
+
+import dev.langchain4j.data.segment.TextSegment;
+import static dev.langchain4j.internal.Utils.randomUUID;
+import dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import dev.langchain4j.store.embedding.EmbeddingStoreIT;
+import java.util.Arrays;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.weaviate.WeaviateContainer;
+
+@Testcontainers
+class LocalGRPCWeaviateEmbeddingStoreIT extends EmbeddingStoreIT {
+
+    @Container
+    static WeaviateContainer weaviate = new WeaviateContainer("semitechnologies/weaviate:latest")
+            .withEnv("QUERY_DEFAULTS_LIMIT", "25")
+            .withEnv("DEFAULT_VECTORIZER_MODULE", "none")
+            .withEnv("AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED", "true")
+            .withEnv("CLUSTER_HOSTNAME", "node1");
+
+    private final EmbeddingStore<TextSegment> embeddingStore = WeaviateEmbeddingStore.builder()
+            .scheme("http")
+            .host(weaviate.getHost())
+            .port(weaviate.getMappedPort(8080))
+            .useGrpc(true)
```

Comment:
It seems that this test still uses HTTP under the hood.
When looking into sources of `WeaviateClient`, it seems that only `batch()` method is aware of grpc, `graphQL()` (which we use) seems to only use HTTP.
Could you please double check if this is the case?

Reply:
Yes you are correct, only batch uses gRPC reading the code from weaviate client

Reply:
Confirmed by Weaviate support: only batch inserts use gRPC

Reply:
Do you know if there is a plan to extend grpc usage for querying as well?
Otherwise I would suggest renaming `useGrpc` into `useGrpcForInserts` or something like that.
Now it is confusing

Reply:
No I don't know about that but it would mean renaming if they change that behaviour. I'd rather stick with weaviate own naming as weaviate doesn't specify that in its configuration

Reply:
@ehsavoie if they do a bad job with their API, it does not mean we should do the same :) Right now it is very confusing.
If they will add an option to use gRPC for reads, we could either add `useGrpcForReads` or `useGrpc` which will apply to both. Do you see a problem with this approach? Thanks

Reply:
No I have no problem with that :)
Iwas just wondering if the host should be configurable as i noticed that on thir cloud it has a different host name than the HTTP one